codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsQuery.java

```diff
@@ -152,4 +155,23 @@ protected void writeErrorMessageCodeDeserializer(GenerationContext context) {
     public void generateProtocolTests(GenerationContext context) {
         AwsProtocolUtils.generateHttpProtocolTests(context);
     }
+
+    @Override
+    public Map<String, ShapeId> getOperationErrors(GenerationContext context, OperationShape operation) {
+        Map<String, ShapeId> errors = new TreeMap<>();
+
+        operation.getErrors().forEach(shapeId -> {
+            Shape errorShape = context.getModel().expectShape(shapeId);
+            String errorName = shapeId.getName(context.getService());
+
+            Optional<AwsQueryErrorTrait> errorShapeTrait = errorShape.getTrait(AwsQueryErrorTrait.class);
+            if (errorShapeTrait.isPresent()) {
+                errors.put(errorShapeTrait.get().getCode(), shapeId);
+            } else {
+                errors.put(errorName, shapeId);
+            }
+        });
+
+        return errors;
```

Comment:
Should this map key, value pair be reversed?  

I would have expected shapeId to be the key, while errorName or code is the value. Is errorName or errorCode guaranteed to be unique for each shape?

Reply:
Error name has to be unique cause it is the shapeId name by default. For query alias’s and whether those conflict that would be a for whether smithy validates that constraint.